src/main/java/org/mockito/internal/creation/DelegatingMethod.java

```diff
@@ -31,7 +31,7 @@ public String getName() {
     }
 
     public Class<?>[] getParameterTypes() {
-        return method.getParameterTypes();
+        return SuspendMethod.trimSuspendParameterTypes(method.getParameterTypes());
```

Comment:
Can we create field on the class that contains trimmed parameter types? This method is invoked frequently (with every invocation on mock), so why trimming every time :)